internal/host/plugin/query.go

```diff
@@ -17,12 +17,35 @@ delete from host_plugin_catalog_secret
  where catalog_id = @catalog_id;
 `
 
+	// FIXME: This needs to take into account sync_interval_seconds being positive
 	setSyncNextRunInQuery = `
 select
-	need_sync as sync_now,
+  case
+    when sync_interval_seconds = -1
+      then
+        false
+      else
+        need_sync
+  end sync_now,
```

Comment:
need_sync is meant to indicate a one time requirement to sync the data.  It feels a little weird to overload the sync interval field to disable all syncs, even the ones triggered by change or potentially explicit requests to sync in the future.  What do you think about having `sync_interval_seconds` be either null (for default) or positive and having a different field to disable syncing all together.  That way when we have an endpoint in the future that can force the resync of a specific set we can return false if the disable sync flag is set and there won't be confusion around why a sync interval field is getting in the way of manually syncing a set.

Reply:
I don't think it's overloading at all -- why would you force a sync of a set that has syncing disabled? What kind of change would force a sync of a set where the user has requested that you _not sync the set_? If there is an endpoint to force a resync, it could error out if sync is disabled, or there could be a `force` flag set to do a one-time override of the disabled synchronization.

Reply:
The reason I feel like it may be overloaded is that the naming of the `sync_interval_seconds` field suggests to me that it is meant to manage how often automatic syncing of a set happens and not decide if the set syncs at all.

I may have a very stable network that I want to connect boundary with so when I create my host set I can set `sync_interval_seconds` to -1 since I don't need to refresh the set of hots since they never change, but because I set -1 I never get the initial set of hosts which should be in my set.

In a slightly different situation I may have an AWS account with instances which never change their characteristics so I don't need automatic syncing, but while I am deciding which sets to create in boundary, I would like to update the filters of 1 set and have them reflected in boundary otherwise I can have a set with a very restricted filter (say to a single instance) but because the `sync_interval_seconds` is set to -1 the set still contains all my aws instances.

Reply:
OK -- I'm fine changing it so it doesn't ignore need_sync -- but I think generally we need to be careful/cautious as to when we force syncs.

Reply:
Updated. Note that we do not actually run this query right now, though.

Reply:
I think this impacts the `setSyncJobQuery` query as well since that query only returns a record when the sync time is either null or > 0.

Reply:
Good point; I'll update that too.